app/frontend/components/common/asset.tsx

```diff
@@ -35,8 +46,12 @@ export const FormattedAssetItem = ({
 }: FormattedAssetItemProps & {
   children: (props: {
     icon: h.JSX.Element
-    formattedAssetName: h.JSX.Element | string
-    formattedAssetIconName: h.JSX.Element
+    formattedHumanReadableLabelType: FormattedHumanReadableLabelType
```

Comment:
I'd consider grouping these "formattedHumanReadableLabel" under a single key which could be a union type with clearly defined fields depending on the type, as it's otherwise pretty hard to follow which of the properties apply when

Reply:
tbh, this passing of formatted components to the children instead of data to to handle the formatting internally seems like an antipattern to me, making the reasoning about what exactly gets rendered a pain, on the other hand, I understand refactoring it would require even more effort without that much added value as I don't expect much more new data/rendering logic to be added in the foreseeable future - maybe data about NFTs but that's what I'd consider the tipping point to refactor/redesign this together with the overall UI more radically

Reply:
Refactored according to the first comment.